lua/cli_flash.lua

```diff
@@ -947,7 +954,14 @@ function exec(aArgs)
 			else
 				-- check if the selected flash is present
 				print("Detecting flash device")
-				fOk = flasher.detect(tPlugin, aAttr, iBus, iUnit, iChipSelect)
+
+				-- Set a specific SPI speed if a value is provided, otherwise detect will set a default
+				-- This also overrides the default maximum speed (A cap of 50MHz still exists as defined in the regdef)
+				local atParameter = {}
+				atParameter.ulInitialSpeed = iSpiKhz or nil

```

Comment:
Do we want to override the initial speed setting which is used during the flash detection? This only makes sense if we are sure that the flash populated on the hardware works at that speed, and all commands sent to the flash during detection also work at that speed.

Reply:
The --spi_khz parameter was made with the assumption the user wants to override the speed and knows what will/will not work.
I am adding a new parameter as mentioned in a comment above so it's an easy change whether the initial speed should be overwritten or not.
I am unsure whether overriding the initial speed or not is the desired behaviour so I'll have to get back to you on this.